### PR TITLE
AK-MIN-459 polkadot v0.9.7

### PR DESCRIPTION
**Pull request by kurotych**
**Description:**

Describe in brief, what was done.

**Changes in storage or API:**

Describe storage structure changes or API changes. Make sure to update the [API docs](https://minterestfinance.atlassian.net/wiki/spaces/MINTEREST/pages/134807557/Minterest+API).

**Checklist**

- [x] Code is covered with tests
- [x] Code is covered with benchmarks
- [x] Fresh benchmark results included

### CODE REVIEW

File: pallets/integration-tests/src/lib.rs [hunk 1]
```diff
@@ -134,8 +136,8 @@ mod tests {
 
 	pub struct WhitelistMembers;
 	impl Contains<u64> for WhitelistMembers {
-		fn sorted_members() -> Vec<u64> {
-			vec![4]
+		fn contains(_: &u64) -> bool {
+			true
 		}
 		#[cfg(feature = "runtime-benchmarks")]
 		fn add(new: &u128) {
```
Comment (lut1k) [inline pallets/integration-tests/src/lib.rs:143]: You can remove `struct WitelistMembers` and impl `Contains`. This struct is not used now.
It is also necessary to remove the same structure from the pallet `risk-manager`.

File: pallets/prices/src/mock.rs [hunk 2]
```diff
@@ -23,9 +23,9 @@ frame_support::construct_runtime!(
 		NodeBlock = Block,
 		UncheckedExtrinsic = UncheckedExtrinsic,
 	{
-		System: frame_system::{Module, Call, Config, Storage, Event<T>},
-		Balances: pallet_balances::{Module, Call, Storage, Config<T>, Event<T>},
-		PricesModule: module_prices::{Module, Storage, Call, Event<T>},
+		System: frame_system::{Pallet, Call, Config, Storage, Event<T>},
+		Balances: pallet_balances::{Pallet, Call, Storage, Config<T>, Event<T>},
+		PricesPallet: module_prices::{Pallet, Storage, Call, Event<T>},
```
Comment (lut1k) [inline pallets/prices/src/mock.rs:28]: why `PricesPallet` but `module_prices`?

File: runtime/src/benchmarking/controller.rs [hunk 3]
```diff
@@ -1,14 +1,12 @@
 use crate::{Balance, Operation, Rate, Runtime, DOT};
 use frame_system::RawOrigin;
 use orml_benchmarking::{runtime_benchmarks, Zero};
-use sp_runtime::FixedPointNumber;
+use sp_runtime::{traits::One, FixedPointNumber};
```
Comment (lut1k) [inline runtime/src/benchmarking/controller.rs:4]: You can remove unused imports from: `controller`, `liquidation_pools`, `minterest_model`, `mnt_token`, `prices`, `risk_manager`.

File: runtime/src/lib.rs [hunk 4]
```diff
@@ -197,6 +197,10 @@ impl frame_system::Config for Runtime {
 	type SystemWeightInfo = ();
 	/// This is used as an identifier of the chain. 42 is the generic substrate prefix.
 	type SS58Prefix = SS58Prefix;
+	/// What to do if the user wants the code set to something. Just use `()` unless you are in
+	/// cumulus.
+	/// TODO MIN-293
```
Comment (lut1k) [inline runtime/src/lib.rs:202]: I didn't find this task on our board.

File: runtime/src/lib.rs [hunk 5]
```diff
@@ -324,8 +338,11 @@ impl pallet_membership::Config<OperatorMembershipInstanceMinterest> for Runtime
 	type SwapOrigin = EnsureRootOrTwoThirdsMinterestCouncil;
 	type ResetOrigin = EnsureRootOrTwoThirdsMinterestCouncil;
 	type PrimeOrigin = EnsureRootOrTwoThirdsMinterestCouncil;
-	type MembershipInitialized = MinterestOracle;
-	type MembershipChanged = MinterestOracle;
+	// TODO These two values was equal to MinterestOracle before updating
+	type MembershipInitialized = (); // MinterestOracle
+	type MembershipChanged = (); // MinterestOracle
```
Comment (lut1k) [inline runtime/src/lib.rs:343]:
```suggestion
	type MembershipInitialized = ();
	type MembershipChanged = MinterestOracle;
```

File: runtime/src/lib.rs [hunk 6]
```diff
@@ -497,8 +531,13 @@ impl orml_oracle::Config<MinterestDataProvider> for Runtime {
 	type OracleValue = Price;
 	type RootOperatorAccountId = ZeroAccountId;
 	type WeightInfo = ();
+	type Members = Members;
```
Comment (lut1k) [inline runtime/src/lib.rs:534]:
```suggestion
	type Members = OperatorMembershipMinterest;
```

File: runtime/src/lib.rs [hunk 7]
```diff
@@ -483,10 +508,19 @@ parameter_types! {
...
+	// TODO MIN-446. Temporary. This is need to pass integration test
+	pub ORACLE1: AccountId = AccountId::from([4u8; 32]);
+	pub OracleMembers: Vec<AccountId> = vec![ORACLE1::get()];
+	pub const MaxHasDispatchedSize: u32 = 100;
+}
+pub struct Members;
+impl SortedMembers<AccountId> for Members {
+	fn sorted_members() -> Vec<AccountId> {
+		OracleMembers::get()
+	}
```
Comment (lut1k) [inline runtime/src/lib.rs:520]: I think you can remove this code

File: runtime/src/lib.rs [hunk 8]
```diff
@@ -558,45 +597,44 @@ construct_runtime!(
 		NodeBlock = opaque::Block,
 		UncheckedExtrinsic = UncheckedExtrinsic
 	{
-		System: frame_system::{Module, Call, Config, Storage, Event<T>},
-		RandomnessCollectiveFlip: pallet_randomness_collective_flip::{Module, Call, Storage},
-		Timestamp: pallet_timestamp::{Module, Call, Storage, Inherent},
+		System: frame_system::{Pallet, Call, Config, Storage, Event<T>},
+		RandomnessCollectiveFlip: pallet_randomness_collective_flip::{Pallet, Call, Storage},
```
Comment (lut1k) [inline runtime/src/lib.rs:601]: It is not entirely clear why this pallet is needed. Compile successfully without it.

**Review by lut1k — APPROVED** [review]

File: runtime/src/lib.rs [hunk 9]
```diff
@@ -324,8 +338,11 @@ impl pallet_membership::Config<OperatorMembershipInstanceMinterest> for Runtime
 	type SwapOrigin = EnsureRootOrTwoThirdsMinterestCouncil;
 	type ResetOrigin = EnsureRootOrTwoThirdsMinterestCouncil;
 	type PrimeOrigin = EnsureRootOrTwoThirdsMinterestCouncil;
-	type MembershipInitialized = MinterestOracle;
+	// TODO These value was equal to MinterestOracle before updating
+	type MembershipInitialized = (); // MinterestOracle
```
Comment (lut1k) [inline runtime/src/lib.rs:342]:
```suggestion
	type MembershipInitialized = ();
```

**Review by lut1k — APPROVED** [review]

**Review by lut1k — APPROVED** [review]

File: runtime/src/benchmarking/minterest_protocol.rs [hunk 10]
```diff
@@ -57,7 +57,7 @@ fn hypothetical_liquidity_setup(borrower: &AccountId, lender: &AccountId) -> Res
 runtime_benchmarks! {
 	{ Runtime, minterest_protocol }
 
-	_ {}
+	// _ {}
```
[the remark this answers is not in the document]
Comment (kurotych) [reply runtime/src/benchmarking/minterest_protocol.rs]: Done

**Review by Apollo50 — APPROVED** [review]

**Review by lut1k — APPROVED** [review]

**Pull request merged**